### PR TITLE
QOLOE-294 Styling search input items

### PR DESCRIPTION
**Pull request by ghazal-salehi**
This resolves the style clashing with https://github.com/qld-gov-au/qgds-bootstrap5/blob/develop/src/scss/qld-type.scss line 102 (top margin for li)
Ran lint also.

### CODE REVIEW

**Review by MitchAnderson93 — APPROVED** [review]

File: src/components/bs5/searchInput/searchInput.scss [hunk 1]
```diff
@@ -98,8 +98,10 @@
 
                     &:hover {
                         background-color: var(--#{$prefix}site-search-suggestions-hover);
-                        margin: 0 -1rem;
-                        padding: 0 1rem;
+                        margin-left: -1rem;
+                        margin-right: -1rem;
+                        padding-right: 1rem;
+                        padding-left: 1rem;
```
Comment (stvp-qld) [inline src/components/bs5/searchInput/searchInput.scss:104]: Hey @ghazal-salehi, is it possible to move these margins away from the hover state?  

Thinking is margins and padding should be consistent across all states, so if we set them on LI's default state and just let it inherit to all others?

**Review by stvp-qld — CHANGES_REQUESTED** [review]

File: src/components/bs5/searchInput/searchInput.scss (hunk 1 above)
Comment (MitchAnderson93) [reply src/components/bs5/searchInput/searchInput.scss:104]: Happy for you guys to take over the styling of this feature

Comment (ghazal-salehi) [reply src/components/bs5/searchInput/searchInput.scss:104]: Thanks Mitch. Please do.

Comment (ghazal-salehi) [reply src/components/bs5/searchInput/searchInput.scss]: Hi @stvp-qld. I removed the padding and margin rules for hover state. Now the search input dropdown style matches the site of the global search in the header.

Comment (MitchAnderson93) [reply src/components/bs5/searchInput/searchInput.scss]: Intended:
Screenshot 2024-06-24 at 11 37 28 am

without margin/padding on hover element:
Screenshot 2024-06-24 at 11 36 33 am

we just have to make sure is reflected elsewhere and can inherit

File: src/components/bs5/searchInput/searchInput.scss [hunk 2]
```diff
@@ -35,7 +35,16 @@
         border-bottom: solid .25rem var(--#{$prefix}site-search-suggestions-hover__border_color);
 
         .suggestions-category {
-            padding: 0 1rem;
+            .suggestion-category-label {
```
Comment (MitchAnderson93) [inline src/components/bs5/searchInput/searchInput.scss:38]: &-label works 
(becomes .suggestion-category-label)

**Review by MitchAnderson93 — DISMISSED** [review]

**Review by duttonw — APPROVED** [review]

**Review by MitchAnderson93 — APPROVED** [review]

**Review by stvp-qld — APPROVED** [review]

**Pull request merged**